src/gleam/erlang/application.gleam

```diff
@@ -0,0 +1,35 @@
+//// An Erlang application is a collection of code that can be loaded into the
+//// Erlang virtual machine and even started and stopped if they define a
+//// start module and supervision tree. Each Gleam package is an Erlang
+//// application.
+
+import gleam/erlang/node.{type Node}
+
+pub type StartType {
+  /// A normal application start.
+  Normal
+  /// The application is distributed and started at the current node because of
+  /// a takeover from Node, either because Erlang's `application:takeover/2`
+  /// function has been called, or because the current node has higher priority
+  /// than the previous node.
+  Takeover(previous: Node)
+  // The application is distributed and started at the current node because of
+  // a failover from the previous node.
+  Failover(previous: Node)
```

Comment:
```suggestion
  /// The application is distributed and started at the current node because of
  /// a failover from the previous node.
  Failover(previous: Node)
```